### PR TITLE
Add PlayerShearBlockEvent

### LINKED ISSUE #3553: Add shear events for carving pumpkins & harvesting honey from beehives/bee nests
**Is your feature request related to a problem? Please describe.**
There is currently no shear event for carving pumpkins or harvesting honey from beehives and bee nests.

**Describe the solution you'd like**
Add a shear event for both scenarios.

### PR DESCRIPTION
**Pull request by JRoy**
Adds an event that is called when a player shears a block.

Current blocks that apply to this event are pumpkins and bee hives/nests.

Closes #3553

### CODE REVIEW

File: Spigot-API-Patches/0222-Add-PlayerShearBlockEvent.patch
Comment (Proximyst) [inline Spigot-API-Patches/0222-Add-PlayerShearBlockEvent.patch]: `io.papermc.paper.event.block`

**Review by Proximyst — DISMISSED** [review]
LGTM; the package we can change when merging.

**Comment by JRoy** [comment]
Package changed to `io.papermc.paper.block` and rebased

File: Spigot-Server-Patches/0592-Add-PlayerShearBlockEvent.patch
Comment (electronicboy) [inline Spigot-Server-Patches/0592-Add-PlayerShearBlockEvent.patch]: Please keep the OBF helper to the first example right above it, 
https://github.com/PaperMC/Paper/blob/master/CONTRIBUTING.md#obfuscation-helpers
This just creates extra noisy diffs

File: Spigot-API-Patches/0231-Add-PlayerShearBlockEvent.patch
Comment (electronicboy) [inline Spigot-API-Patches/0231-Add-PlayerShearBlockEvent.patch]: This shouldn't be nullable, ex; making blocks which don't drop items, drop items

**Review by electronicboy — CHANGES_REQUESTED** [review]

File: Spigot-API-Patches/0231-Add-PlayerShearBlockEvent.patch
Comment (JRoy) [reply Spigot-API-Patches/0231-Add-PlayerShearBlockEvent.patch]: No idea why I made this nullable as it's never null 0_o

**Review by Proximyst — DISMISSED** [review]

File: Spigot-Server-Patches/0592-Add-PlayerShearBlockEvent.patch
Comment (electronicboy) [reply Spigot-Server-Patches/0592-Add-PlayerShearBlockEvent.patch]: Looking back at this, this was the "wrong" change, if you just want an OBFHELPER, it should be added to Block, so that that `a` method is mapped (ideally in MCUtils for later calls) for anything that wants to call it, as this is not so much an obfhelper here but a behavioral change to the method in the name of an obfhelper, beyond that, looks ok

**Review by electronicboy — DISMISSED** [review]
See comments, provisionally approved

**Comment by MiniDigger** [comment]
@limbo-app ready

**Review by limbo-app — APPROVED** [review]
Pull request automatically approved for merging - a core team member has previously approved this pull request.

**Comment by Cryptite** [comment]
Does this include Berry Bushes? One effectively "shears it" for berries.

**Comment by Machine-Maker** [comment]
I don't think it should. You don't use shears on the bush.You can probably figure that out via PlayerInteract events

**Pull request merged**
